src/redux/movies.ts

```diff
@@ -43,34 +43,54 @@ const fetchMovies = async ({
   ]
     .filter(Boolean)
     .join('&');
-  const response = await fetch(`http://localhost:4000/movies?${queryParams}`);
+  const response = await fetch(`${URL}/movies?${queryParams}`);
 
   return await response.json();
 };
 
 export const addMovieAction = createAsyncThunk(
   'addMovie',
   async (movie: Movie) => {
-    await fetch('http://localhost:4000/movies', {
+    await fetch(`${URL}/movies`, {
       method: 'POST',
       body: JSON.stringify(movie),
       headers: { 'Content-Type': 'application/json;charset=UTF-8' },
     });
-    const response = await fetch('http://localhost:4000/movies');
+    const response = await fetch(`${URL}/movies`);
 
     return await response.json();
   },
 );
+enum Method {
+  'Put' = 'PUT',
+  'Get' = 'GET',
+  'Post' = 'POST',
+  'Delete' = 'DELETE',
+}
+interface FetchDataParams {
+  url: string;
+  method?: Method;
+  bodyObj?: any;
+  headers?: HeadersInit;
+}
+const fetchJson = async ({ url, method, bodyObj }: FetchDataParams) => {
+  return await fetch(url, {
+    method,
+    body: JSON.stringify(bodyObj),
+    headers: { 'Content-Type': 'application/json;charset=UTF-8' },
+  });
+};
 
 export const updateMovieAction = createAsyncThunk(
   'updateMovie',
   async (movie: Movie) => {
-    await fetch('http://localhost:4000/movies', {
-      method: 'PUT',
-      body: JSON.stringify(movie),
-      headers: { 'Content-Type': 'application/json;charset=UTF-8' },
+    await fetchJson({
+      url: `${URL}/movies`,
+      method: Method.Put,
+      bodyObj: movie,
     });
-    const response = await fetch('http://localhost:4000/movies');
+
+    const response = await fetch(`${URL}/movies`);
```

Comment:
When you implement some custom wrapper above native function (`fetchJson` in our case), you should always use it, we can reuse it here as well